minifi-nar-bundles/minifi-provenance-reporting-bundle/minifi-provenance-reporting-task/src/main/test/java/org/apache/nifi/minifi/provenance/reporting/TestProvenanceReportingTask.java

```diff
@@ -0,0 +1,218 @@
+                    e.printStackTrace();
+                    Assert.fail(e.toString());
+                }
+
+                return client;
+            }
+        };
+
+        final List<ProvenanceEventRecord> events = new ArrayList<>();
+        events.add(event);
+
+        final Map<PropertyDescriptor, String> properties = new HashMap<>();
+        for (final PropertyDescriptor descriptor : task.getSupportedPropertyDescriptors()) {
+            properties.put(descriptor, descriptor.getDefaultValue());
+        }
+        properties.put(ProvenanceReportingTask.BATCH_SIZE, "1000");
+
+        final ReportingContext context = Mockito.mock(ReportingContext.class);
+        Mockito.when(context.getStateManager())
+                .thenReturn(new MockStateManager(task));
+        Mockito.doAnswer(new Answer<PropertyValue>() {
+            @Override
+            public PropertyValue answer(final InvocationOnMock invocation) throws Throwable {
+                final PropertyDescriptor descriptor = invocation.getArgumentAt(0, PropertyDescriptor.class);
+                return new MockPropertyValue(properties.get(descriptor), null);
+            }
+        }).when(context).getProperty(Mockito.any(PropertyDescriptor.class));
+
+        final EventAccess eventAccess = Mockito.mock(EventAccess.class);
+        Mockito.doAnswer(new Answer<List<ProvenanceEventRecord>>() {
+            @Override
+            public List<ProvenanceEventRecord> answer(final InvocationOnMock invocation) throws Throwable {
+                final long startId = invocation.getArgumentAt(0, long.class);
+                final int maxRecords = invocation.getArgumentAt(1, int.class);
+
+                final List<ProvenanceEventRecord> eventsToReturn = new ArrayList<>();
+                for (int i = (int) Math.max(0, startId); i < (int) (startId + maxRecords) && i < events.size(); i++) {
+                    eventsToReturn.add(events.get(i));
+                }
+                return eventsToReturn;
+            }
+        }).when(eventAccess).getProvenanceEvents(Mockito.anyLong(), Mockito.anyInt());
+
+        Mockito.when(context.getEventAccess()).thenReturn(eventAccess);
+
+        final ComponentLog logger = Mockito.mock(ComponentLog.class);
+        final ReportingInitializationContext initContext = Mockito.mock(ReportingInitializationContext.class);
+        Mockito.when(initContext.getIdentifier()).thenReturn(UUID.randomUUID().toString());
+        Mockito.when(initContext.getLogger()).thenReturn(logger);
+
+
+        task.initialize(initContext);
+        task.onTrigger(context);
+
+        assertEquals(1, dataSent.size());
+        final String msg = new String(dataSent.get(0), StandardCharsets.UTF_8);
+        System.out.println(msg);
+    }
+
+    public static FlowFile createFlowFile(final long id, final long fileSize, final Map<String, String> attributes) {
+        final Map<String, String> attrCopy = new HashMap<>(attributes);
+
+        return new FlowFile() {
```

Comment:
Can likely use MockFlowFile instead of a new implementation.  Doesn't appear as though filesize is being used and the mock will allow setting of both the attributes and id.